notify/impl.go

```diff
@@ -759,8 +763,10 @@ func (n *Hipchat) Notify(ctx context.Context, as ...*types.Alert) (bool, error)
 		data     = n.tmpl.Data(receiverName(ctx, n.logger), groupLabels(ctx, n.logger), as...)
 		tmplText = tmplText(n.tmpl, data, &err)
 		tmplHTML = tmplHTML(n.tmpl, data, &err)
-		url      = fmt.Sprintf("%sv2/room/%s/notification?auth_token=%s", n.conf.APIURL, n.conf.RoomID, n.conf.AuthToken)
+		// n.conf.APIURL is already checked at configuration load time.
+		url, _ = url.Parse(n.conf.APIURL.String())
```

Comment:
I am a bit confused here, why not just `n.conf.APIURL` is this not already a URL?

Reply:
`n.conf.APIURL` is a pointer so I didn't want to modify it.

Reply:
you could also do

```go
u := *c.conf.APIURL
u.Path += fmt.Sprintf("v2/room/%s/notification?auth_token=%s", n.conf.RoomID, n.conf.AuthToken)
// ...
resp, err := ctxhttp.Post(ctx, c, u.String(), contentTypeJSON, &buf)
```

Reply:
since we've assigned something to variable name `url`, we've shadowed the reference to the package `net/url` in this scope. Having url's named `u` is annoying, but at some point someone will want to use `net/url` later in this package..